code/mathematical_algorithms/src/exponentiation_power/exponentiation_by_squaring/exponentiation_by_squaring.c

```diff
@@ -0,0 +1,30 @@
+#include <stdio.h>
+
+int 
+power(int num, int exp)
+{
+	if (exp == 0)
+		return (1);
+	if (exp == 1) 
+		return (num);
+
+	int temp = power(num, exp / 2);
+	temp *= temp;
+	if (exp % 2 ==1 )
+		temp *= num;
+
+	return (temp);
```

Comment:
Why do you put parentheses here?

Reply:
Style Guide says parentheses are required for return values.Should I remove them if they are variables?

Reply:
Sorry for the opinion, please ignore it.
But I don't know why the guide is😅, I think the parentheses is ugly and unnecessary.